main.go

```diff
@@ -2,30 +2,98 @@ package main
 
 import "fmt"
 
-//Реализуйте функцию, которая принимает карту и пороговое значение. Возвращайте новую карту, содержащую только те элементы, где значение больше порога.
+//Напишите функцию, которая создаёт "обратную" карту. Например, из карты вида {"a": 1, "b": 2, "c": 2} сделайте карту вида {1: ["a"], 2: ["b", "c"]}
 
-func returnMap(massiv map[string]float32, n float32) map[string]float32 {
-	slice := map[string]float32{}
+func returnMap(massiv map[string]float32) (map[float32]string, error) {
+	var n float32
+	var s string
+	slice := map[float32]string{}
 
 	for i := range massiv {
-		for massiv[i] > n {
-			slice[i] = massiv[i]
-			break
+		n = massiv[i]
+		s = i
+		slice[n] = s
+	}
+	return slice, nil
+}
```

Comment:
func invertMap(input map[string]int) (map[int][]string, error) {
```suggestion
}
```
	result := make(map[int][]string)

	for k, v := range input {
                if _, ok := result[v]; !ok { result[v] = make([]string, 0, 1) } // если слайса в мапе нет, то создаём его
		result[v] = append(result[v], k) // аппендим к слайсу новое значение
	}

	return result, nil
}